core/store.go

```diff
@@ -140,6 +145,23 @@ func DelByPtr(ptr unsafe.Pointer) bool {
 	return false
 }
 
+// List all keys in the store by given pattern
+func Keys(p string) []string {
+	storeMutex.RLock()
+	defer storeMutex.RUnlock()
+	keypoolMutex.RLock()
+	defer keypoolMutex.RUnlock()
+
+	keys := make([]string, 0)
```

Comment:
We might be able to do a small optimization here, if the pattern is an empty string, then preallocate the slice's capacity to be the same as the length of the keypool (to avoid frequent allocations)